go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -532,10 +536,339 @@ func (c *cnxn) GetInfo(ctx context.Context, infoCodes []adbc.InfoCode) (array.Re
+	if err != nil {
+		return nil, adbcFromFlightStatus(err)
+	}
+
+	rdr, err := c.readInfo(ctx, schema_ref.Catalogs, info)
+	if err != nil {
+		return nil, adbcFromFlightStatus(err)
+	}
+	defer rdr.Release()
+
+	catalogToDbSchemas, err := c.getObjectsDbSchemas(ctx, depth, catalog, dbSchema)
+	if err != nil {
+		return nil, err
+	}
+
+	tableLookup, err := c.getObjectsTables(ctx, depth, catalog, dbSchema, tableName, columnName, tableType)
+	if err != nil {
+		return nil, err
+	}
+
+	catalogNameBuilder := builder.Field(0).(*array.StringBuilder)
+	catalogDbSchemasBuilder := builder.Field(1).(*array.ListBuilder)
+	catalogDbSchemasItems := catalogDbSchemasBuilder.ValueBuilder().(*array.StructBuilder)
+	dbSchemaNameBuilder := catalogDbSchemasItems.FieldBuilder(0).(*array.StringBuilder)
+	dbSchemaTablesBuilder := catalogDbSchemasItems.FieldBuilder(1).(*array.ListBuilder)
+	dbSchemaTablesItems := dbSchemaTablesBuilder.ValueBuilder().(*array.StructBuilder)
+	tableNameBuilder := dbSchemaTablesItems.FieldBuilder(0).(*array.StringBuilder)
+	tableTypeBuilder := dbSchemaTablesItems.FieldBuilder(1).(*array.StringBuilder)
+	tableColumnsBuilder := dbSchemaTablesItems.FieldBuilder(2).(*array.ListBuilder)
+	tableColumnsItems := tableColumnsBuilder.ValueBuilder().(*array.StructBuilder)
+	columnNameBuilder := tableColumnsItems.FieldBuilder(0).(*array.StringBuilder)
+	ordinalPositionBuilder := tableColumnsItems.FieldBuilder(1).(*array.Int32Builder)
+	remarksBuilder := tableColumnsItems.FieldBuilder(2).(*array.StringBuilder)
+	xdbcDataTypeBuilder := tableColumnsItems.FieldBuilder(3).(*array.Int16Builder)
+	xdbcTypeNameBuilder := tableColumnsItems.FieldBuilder(4).(*array.StringBuilder)
+	xdbcColumnSizeBuilder := tableColumnsItems.FieldBuilder(5).(*array.Int32Builder)
+	xdbcDecimalDigitsBuilder := tableColumnsItems.FieldBuilder(6).(*array.Int16Builder)
+	xdbcNumPrecRadixBuilder := tableColumnsItems.FieldBuilder(7).(*array.Int16Builder)
+	xdbcNullableBuilder := tableColumnsItems.FieldBuilder(8).(*array.Int16Builder)
+	xdbcColumnDefBuilder := tableColumnsItems.FieldBuilder(9).(*array.StringBuilder)
+	xdbcSqlDataTypeBuilder := tableColumnsItems.FieldBuilder(10).(*array.Int16Builder)
+	xdbcDatetimeSubBuilder := tableColumnsItems.FieldBuilder(11).(*array.Int16Builder)
+	xdbcCharOctetLengthBuilder := tableColumnsItems.FieldBuilder(12).(*array.Int32Builder)
+	xdbcIsNullableBuilder := tableColumnsItems.FieldBuilder(13).(*array.StringBuilder)
+	xdbcScopeCatalogBuilder := tableColumnsItems.FieldBuilder(14).(*array.StringBuilder)
+	xdbcScopeSchemaBuilder := tableColumnsItems.FieldBuilder(15).(*array.StringBuilder)
+	xdbcScopeTableBuilder := tableColumnsItems.FieldBuilder(16).(*array.StringBuilder)
+	xdbcIsAutoincrementBuilder := tableColumnsItems.FieldBuilder(17).(*array.BooleanBuilder)
+	xdbcIsGeneratedcolumnBuilder := tableColumnsItems.FieldBuilder(18).(*array.BooleanBuilder)
+	tableConstraintsBuilder := dbSchemaTablesItems.FieldBuilder(3).(*array.ListBuilder)
```

Comment:
i'm horrified but i don't think there's a better way haha. Maybe hide this behind a struct and then implement methods on that struct so we can pass it around and better modularize this huge function?

Reply:
Yeah, and I can probably break up those loops a bit at that point

Reply:
Refactored into a separate struct, with a few methods instead of one giant method, and rearranged things to unnest the loops